clients/admin-ui/src/features/privacy-requests/configuration/TwilioEmailConfiguration.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from "react";
+
+import { CustomTextInput } from "~/features/common/form/inputs";
+import { isErrorResult } from "~/features/common/helpers";
+import { useAlert, useAPIHelper } from "~/features/common/hooks";
+import {
+  useCreateMessagingConfigurationMutation,
+  useCreateMessagingConfigurationSecretsMutation,
+  useGetMessagingConfigurationDetailsQuery,
+} from "~/features/privacy-requests/privacy-requests.slice";
+
+const TwilioEmailConfiguration = () => {
+  const [configurationStep, setConfigurationStep] = useState("");
+  const { successAlert } = useAlert();
+  const { handleError } = useAPIHelper();
+  const { data: messagingDetails } = useGetMessagingConfigurationDetailsQuery({
+    type: "twilio_email",
+  });
+  const [createMessagingConfiguration] =
+    useCreateMessagingConfigurationMutation();
+  const [createMessagingConfigurationSecrets] =
+    useCreateMessagingConfigurationSecretsMutation();
+
+  const handleTwilioEmailConfiguration = async (value: { email: string }) => {
+    const result = await createMessagingConfiguration({
+      type: "twilio_email",
+      details: {
+        twilio_email_from: value.email,
+      },
+    });
+
+    if (isErrorResult(result)) {
+      handleError(result.error);
+    } else {
+      successAlert(
+        `Twilio email successfully updated. You can now enter your security key.`
+      );
+      setConfigurationStep("configureTwilioEmailSecrets");
+    }
+  };
+
+  const handleTwilioEmailConfigurationSecrets = async (value: {
+    api_key: string;
+  }) => {
+    const result = await createMessagingConfigurationSecrets({
+      twilio_api_key: value.api_key,
+    });
+
+    if (isErrorResult(result)) {
+      handleError(result.error);
+    } else {
+      successAlert(`Twilio email secrets successfully updated.`);
+      setConfigurationStep("configureTwilioEmailSecrets");
+    }
+  };
+
+  const initialValues = {
+    email: messagingDetails?.email ?? "",
+  };
+
+  const initialAPIKeyValues = {
+    api_key: messagingDetails?.key ?? "",
+  };
+
+  return (
+    <Box>
+      <Heading fontSize="md" fontWeight="semibold" mt={10}>
+        Twilio Email messaging configuration
+      </Heading>
+      <Stack>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={handleTwilioEmailConfiguration}
+        >
+          {({ isSubmitting, resetForm }) => (
+            <Form>
+              <Stack mt={5} spacing={5}>
+                <CustomTextInput
+                  name="email"
+                  label="Email"
+                  placeholder="Enter email"
+                />
+              </Stack>
+              <Box mt={10}>
+                <Button
+                  onClick={() => resetForm()}
+                  mr={2}
+                  size="sm"
+                  variant="outline"
+                >
+                  Cancel
+                </Button>
+                <Button
+                  isDisabled={isSubmitting}
```

Comment:
These buttons could probably all benefit from `isLoading` as well. (Guessing it doesn't stand out because the BE isn't running yet.)